go/vt/vttablet/tabletmanager/action_agent.go

```diff
@@ -849,3 +870,12 @@ func (agent *ActionAgent) withRetry(ctx context.Context, description string, wor
 		}
 	}
 }
+
+func validateDemoteMasterType(tabletTypeParam string) (*topodatapb.TabletType, error) {
+	tabletType, err := topoproto.ParseTabletType(tabletTypeParam)
+	if err != nil && (tabletType != topodatapb.TabletType_SPARE && tabletType != topodatapb.TabletType_REPLICA) {
```

Comment:
It looks like you meant for that first `&&` to be an `||`. However, even then I don't think it will do the right thing. That might return a nil `err` when there's actually a problem.

I think what we want is:

```go
if err != nil {
  return topodatapb.TabletType_UNKNOWN, err
}
if tabletType != topodatapb.TabletType_SPARE && tabletType != topodatapb.TabletType_REPLICA {
  return topodatapb.TabletType_UNKNOWN, fmt.Errorf("invalid demote_master_type %v; can only be REPLICA or SPARE", tabletType)
}
return tabletType, nil
```

Reply:
oh, fixed. Using unknown tablet type as suggested.